flocker/control/httpapi.py

```diff
@@ -853,6 +865,74 @@ def replace_configuration(self, applications, deployment):
         except ConfigurationError as e:
             raise make_bad_request(code=BAD_REQUEST, description=unicode(e))
 
+    @app.route("/configuration/leases", methods=['GET'])
+    # This can stop being private as part of FLOC-2741:
+    @private_api
+    @structured(
+        inputSchema={},
+        outputSchema={
+            '$ref':
+            '/v1/endpoints.json#/definitions/list_leases'
+        },
+        schema_store=SCHEMAS
+    )
+    def list_leases(self):
+        """
+        List the current leases in the configuration.
+        """
+        now = datetime.fromtimestamp(self.clock.seconds(), UTC)
+        result = []
+        for lease in self.persistence_service.get().leases.values():
+            result.append(lease_to_json(lease, now))
+        return result
+
+    @app.route("/configuration/leases/<dataset_id>", methods=['DELETE'])
+    # This can stop being private as part of FLOC-2741:
+    @private_api
+    @user_documentation(
+        u"""
+        This will release a lease on a dataset, allowing it to be moved
+        elsewhere, deleted and otherwise modified.
+        """,
```

Comment:
:question: Do you think we should clarify in the reference docs that this doesn't do anything to the dataset the lease concerns?